dapps/src/api/time.rs

```diff
@@ -169,40 +192,39 @@ impl<F: Fetch, T: TimeProvider> TimeChecker<F, T> {
 mod tests {
 	use std::sync::Arc;
 	use std::cell::RefCell;
-	use fetch::{self, Fetch};
-	use futures::{self, Future};
-	use futures::future::FutureResult;
-	use super::{TimeProvider, TimeChecker, Error};
-	use util::Mutex;
+	use std::time::Instant;
+	use time::Duration;
+	use futures::{self, BoxFuture, Future};
+	use super::{Ntp, TimeChecker, Error};
+	use util::{Mutex, RwLock};
 
 	#[derive(Clone)]
-	struct FakeFetch(bool, Arc<Mutex<u64>>);
-	impl Fetch for FakeFetch {
-		type Result = FutureResult<fetch::Response, fetch::Error>;
-		fn new() -> Result<Self, fetch::Error> where Self: Sized { Ok(FakeFetch(false, Default::default())) }
-		fn fetch_with_abort(&self, url: &str, _abort: fetch::Abort) -> Self::Result {
-			assert_eq!(url, "https://time.parity.io/api");
-			let mut val = self.1.lock();
-			*val = *val + 1;
-			if self.0 {
-				futures::future::ok(fetch::Response::not_found())
-			} else {
-				let data = ::std::io::Cursor::new(b"1".to_vec());
-				futures::future::ok(fetch::Response::from_reader(data))
-			}
+	struct FakeNtp(RefCell<Vec<Duration>>, Arc<Mutex<u64>>);
+	impl FakeNtp {
+		fn new() -> FakeNtp {
+			FakeNtp(
+				RefCell::new(vec![Duration::milliseconds(150)]),
+				Arc::new(Mutex::new(0)))
 		}
 	}
-	#[derive(Clone)]
-	struct FakeTime(RefCell<Vec<i64>>);
-	impl TimeProvider for FakeTime {
-		fn new() -> Self where Self: Sized { FakeTime(RefCell::new(vec![150, 0])) }
-		fn utc_timestamp_millis(&self) -> i64 {
-			self.0.borrow_mut().pop().expect("Expecting only two calls to utc_timestamp_millis.")
+
+	impl Ntp for FakeNtp {
+		fn drift(&self) -> BoxFuture<Duration, Error> {
+			let mut val = self.1.lock();
+			*val = *val + 1;
+			futures::future::ok(self.0.borrow_mut().pop().expect("Expecting only one call to now().")).boxed()
```

Comment:
since `self.0` can be modified outside I would rather write something like: "Unexpected call to `drift()` instead.